npm/qsharp/src/shared/circuit.ts

```diff
@@ -56,32 +56,52 @@ export interface Qubit {
 }
 
 /**
- * Represents an operation and the registers it acts on.
+ * Base type for operations.
  */
-export interface Operation {
+export interface BaseOperation {
   /** Gate label. */
   gate: string;
   /** Formatted gate arguments. */
   args?: string[];
   /** Nested operations within this operation. */
   children?: ComponentGrid;
-  /** Whether gate is a measurement operation. */
-  isMeasurement?: boolean;
+
+  /** Custom data attributes to attach to gate element.
+  Note that this is never written to file, so it is not part of the circuit schema */
+  dataAttributes?: DataAttributes;
+
   /** Whether gate is a conditional operation. */
   isConditional?: boolean;
-  /** Whether gate is an adjoint operation. */
-  isAdjoint?: boolean;
-  /** Control registers the gate acts on. */
-  controls?: Register[];
-  /** Target registers the gate acts on. */
-  targets: Register[];
   /** Specify conditions on when to render operation. */
   conditionalRender?: ConditionalRender;
-  /** Custom data attributes to attach to gate element.
-  Note that this is never written to file, so it is not part of the circuit schema */
-  dataAttributes?: DataAttributes;
 }
 
+export interface Measurement extends BaseOperation {
+  /** Discriminator for the Operation type */
+  kind: "Measurement";
```

Comment:
I'm used to seeing camelcased names in these kinds of values. Not sure if there's an "official" convention though

Reply:
done